aten/src/ATen/core/jit_type.h

```diff
@@ -256,6 +262,9 @@ struct CAFFE2_API UndefinedTensorType : public Type {
     return rhs.kind() == kind();
   }
   bool isSubtypeOf(const TypePtr rhs) const override {
+    if(auto rhs_ = rhs->cast<OptionalType>()) {
```

Comment:
Rather than copypasta, just call the super class:
```
return rhs->kind() == TypeKind::DynamicType ||
           rhs->kind() == TypeKind::UndefinedTensorType ||
           DynamicType::isSubtypeOf(rhs); 
```

Reply:
hmmm actually `isSubtypeOf` is not a static member, and UndefinedTensor/Tensor/CompleteTensor/Dynamic is not subclass/superclass relationship. Do we want to either make `isSubtypeOf` a static method, or find a way to make a subclassing hierarchy for those tensor related types?

Reply:
Unfortunately, a method cannot be both `virtual` and `static`, there might be a solution if we want to go towards the static `isSubtypeOf` approach as indicated [here](https://stackoverflow.com/questions/34222703/how-to-override-static-method-of-template-class-in-derived-class)

Reply:
It is fine to only repeat this for things that are not in the DynamicClass hierarchy, but CompleteTensor and Tensor are subclasses of Dynamic and it should not be repeated in them.

Reply:
It is not a super method in `class UndefinedTensorType`, because `UndefinedTensorType` and `DynamicType` are not inheritance relationship. In `UndefinedTensorType`, we can only call `Type::isSubtypeOf(rhs)`

Reply:
Yeah that sounds good